### PR TITLE
refactor: saved search alerts (approach #2)

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Refactor**

Review app: https://saved-search-alert.artsy.net/

### Changes
This PR includes the changes that were suggested by Matt [here](https://github.com/artsy/force/pull/10307#discussion_r904977276)

### CODE REVIEW

File: src/v2/Components/SavedSearchAlert/types.ts [hunk 1]
```diff
@@ -21,27 +21,38 @@ export interface SearchCriteriaAttributes {
...
 }
 
+export interface SavedSearchEntityCriteria {
+  displayValue: string
+  value: string
+}
+
+export type SavedSearchDefaultCriteria = Record<
+  string, // TODO: Specify allowed keys
+  SavedSearchEntityCriteria | SavedSearchEntityCriteria[]
```
Comment (dimatretyak) [inline src/v2/Components/SavedSearchAlert/types.ts:47]: Note that the value for default criteria can be single element or an array of elements. This is necessary in order to extract search criteria correctly when an array of elements is expected (for example, as for `artistIDs` field)

**Single element**
```javascript
defaultCriteria: {
  offerable: {
    displayValue: "Offerable",
    value: "offerable",
  },
}
```

**Multiple elements**
```javascript
defaultCriteria: {
  artistIDs: [
    {
      displayValue: artist.name ?? "",
      value: artist.internalID,
    },
  ],
}
```

File: src/v2/Apps/Artist/Routes/WorksForSale/Components/ArtistArtworkFilter.tsx [hunk 2]
```diff
@@ -98,7 +115,12 @@ const ArtistArtworkFilter: React.FC<ArtistArtworkFilterProps> = props => {
           aggregations: ["TOTAL"],
         }}
         enableCreateAlert
-        savedSearchEntity={savedSearchEntity}
+        renderFilterPills={() => (
+          <ActiveFilterPills defaultPills={defaultPills} />
+        )}
+        renderCreateAlertButton={() => (
+          <DefaultCreateAlertButton savedSearchEntity={savedSearchEntity} />
+        )}
```
Comment (dimatretyak) [inline src/v2/Apps/Artist/Routes/WorksForSale/Components/ArtistArtworkFilter.tsx:123]: Please note that we now need to duplicate some artist info a bit and pass it for pills and create alert flow separately

**Review by mdole — DISMISSED** [review]
really nice work @dimatretyak. I think this is going to make it much easier to expand alerts to other grids and to maintain this code in the future!

File: src/v2/Components/SavedSearchAlert/Utils/extractPills.ts [hunk 3]
```diff
@@ -189,17 +189,63 @@ export const extractPillsFromCriteria = ({
...
-export const extractArtistPills = (
-  defaultArtists: SavedSearchEntityArtist[] = []
-): FilterPill[] => {
-  return defaultArtists.map(artist => {
-    return {
-      isDefault: true,
-      value: artist.id,
-      displayValue: artist.name,
-      field: "artistIDs",
+export const extractPillsFromDefaultCriteria = (
```
[the remark this answers is not in the document]
Comment (dimatretyak) [reply src/v2/Components/SavedSearchAlert/Utils/extractPills.ts:192]: thank you!

**Review by nickskalkin — APPROVED** [review]
Looks good to me! 👍

**Review by gkartalis — APPROVED** [review]
🎉 🎉 🎉 🎉

**Pull request merged**